dataland-backend/src/test/kotlin/org/dataland/datalandbackend/services/ApiKeyUtilityTest.kt

```diff
@@ -0,0 +1,107 @@
+package org.dataland.datalandbackend.services
+
+import org.dataland.datalandbackendutils.apikey.ApiKeyUtility
+import org.dataland.datalandbackendutils.apikey.ParsedApiKey
+import org.dataland.datalandbackendutils.exceptions.ApiKeyFormatException
+import org.dataland.datalandbackendutils.utils.EncodingUtils
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+
+class ApiKeyUtilityTest {
+
+    private val apiKeyUtility = ApiKeyUtility()
+
+    private val testKeycloakUserId = "c5ef10b1-de23-4a01-9005-e62ea226ee83"
+    private val testApiKeyBase64EncodedKeycloakUserId = "YzVlZjEwYjEtZGUyMy00YTAxLTkwMDUtZTYyZWEyMjZlZTgz"
+    private val testApiKeySecret = "da030a8c290f43a022bdb3da59f6ee4d7b5e290997fbdb3a22f1a7ce5a1a51c87f7666a3476a950b"
+    private val testApiKeyCrc32Value = "465063745"
+
+    private fun getCrc(keycloakUserIdBase64Encoded: String, apiKeySecret: String): String {
+        val parsedApiKeyWithoutCrc32Value = keycloakUserIdBase64Encoded + "_" + apiKeySecret
+
+        return EncodingUtils.calculateCrc32Value(
+            parsedApiKeyWithoutCrc32Value.toByteArray()
+        ).toString()
+    }
+
+    private fun prevalidateBrokenApiKeysAndAssertThrownMessages(
+        mapOfBrokenApiKeysAndExpectedMessages:
+            Map<String, String>
```

Comment:
Die map hier bringt eigentlich keinen Mehrwert, man könnte auch einfach zwei input Parameter nehmen und es würde nur einen Test ändern aber den overhead der map komplett entfernen, das müsste eigentlich insgesamt kürzer sein

Reply:
Danke für den Hinweis, ich schaue mal rein.
Die Idee der map ist eigentlich auch nur entstanden, weil ich an einer Stelle mehrere cases abfrühstücken wollte.
Aber könnte ich ja auch über forEach machen.

Reply:
Fixed.   Funktion akzeptiert jetzt zwei Parameter, und wird für den einen Test einfach in der loop zweimal ausgeführt.